src/wrappers/themis/Obj-C/objcthemis/skeygen.m

```diff
@@ -80,3 +80,24 @@ - (TSErrorType)generateKeys {
 }
 
 @end
+
+NSData* __nullable TSGenerateSymmetricKey()
+{
+    TSErrorType result;
+    NSMutableData *key;
+    size_t keyLength = 0;
+
+    result = (TSErrorType) themis_gen_sym_key(NULL, &keyLength);
+    if (result != TSErrorTypeBufferTooSmall) {
+        return nil;
+    }
+
+    key = [NSMutableData dataWithLength:keyLength];
+
+    result = (TSErrorType) themis_gen_sym_key(key.mutableBytes, &keyLength);
+    if (result != TSErrorTypeSuccess) {
```

Comment:
```suggestion
    if (result != TSErrorTypeSuccess) {
    [key resetBytesInRange:NSMakeRange(0, [key length])];
```

Reply:
~Also, let's add tests here: https://github.com/cossacklabs/themis/tree/master/tests/objcthemis/objthemis~

Found tests :)

Reply:
> `[key resetBytesInRange:NSMakeRange(0, [key length])];`

I'd think this will be excessive when [these suggestion](https://github.com/cossacklabs/themis/pull/565#discussion_r354419921) is implemented in Themis Core. It will ensure that if themis_gen_sym_key() fails then output buffer does not contain anything useful or sensitive. That way we will not need to add additional wiping to every single wrapper individually.

Reply:
agree.

another reason I suggested it for iOS – I don't know how to zeroing bytes in every other language :D